src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkflowComponent.scala

```diff
@@ -244,6 +247,19 @@ trait WorkflowComponent {
       }
     }
 
+    def batchUpdateStatusAndExecutionServiceKey(workflows: Seq[WorkflowRecord], newStatus: WorkflowStatus, execServiceId: ExecutionServiceId): ReadWriteAction[Int] = {
+      if (workflows.isEmpty) {
+        DBIO.successful(0)
+      } else {
+        UpdateWorkflowStatuAndExecutionIdRawSql.actionForWorkflowRecs(workflows, newStatus, execServiceId) flatMap { rows =>
+          if (rows.head == workflows.size)
+            DBIO.successful(workflows.size)
+          else
+            throw new RawlsConcurrentModificationException(s"could not update ${workflows.size - rows.head} workflows because their record version(s) have changed")
```

Comment:
this should be DBIO.failed(exception) instead of throw new exception but really you don't need the flatMap. Use map instead:
map { rows => if (rows.head == workflows.size) workflows.size else throw new exception }